### ISSUE TITLE
How to test?

### ISSUE THREAD

**Issue by Vieeo** [mcordts/cityscapesScripts#143]
I have got 19 classes color png of testset,I want to know mIoU, how should I do ? Or how can I convert 19 cls into 34 cls to offical website?

**Comment by Linfan880** [comment]
I have got 19 classes color png of testset,I want to know mIoU, how should I do ? Or how can I convert 19 cls into 34 cls to offical website?

**Comment by Jess1989** [comment]
> I have got 19 classes color png of testset,I want to know mIoU, how should I do ? Or how can I convert 19 cls into 34 cls to offical website?

have you solved this problem?

**Comment by Jess1989** [comment]
> I have got 19 classes color png of testset,I want to know mIoU, how should I do ? Or how can I convert 19 cls into 34 cls to offical website?

have you solved this problem?

**Comment by Lature-Yang** [comment]
have you solved this problem?

**Comment by Jess1989** [comment]
> 

yes

**Comment by Lature-Yang** [comment]
and how? Could you share  the way to deal with it ？

**Comment by Jess1989** [comment]
> and how? Could you share the way to deal with it ？

give me ur email

**Comment by Lature-Yang** [comment]
lopeture@gmail.com, Thanks a lot.

**Comment by xiaodaomingye459** [comment]
> > and how? Could you share the way to deal with it ？
> 
> give me ur email

755435459@qq.com, Thanks a lot

**Comment by Chihanxx** [comment]
> > 
> 
> 是的

Hi, Could you share the way to deal with it ？my email is 752103930@qq.com,Thanks a lot.

**Comment by Chihanxx** [comment]
> > and how? Could you share the way to deal with it ？
> 
> give me ur email

Hi, Could you share the way to deal with it ？my email is [752103930@qq.com](mailto:752103930@qq.com),Thanks a lot.

**Comment by Jess1989** [comment]
what do u mean? do u mean how to translate to labelIDs or how to submit on cityscapes web?
----- 原始邮件 -----
发件人：KrisLi ***@***.***>
收件人：mcordts/cityscapesScripts ***@***.***>
抄送人：Jess1989 ***@***.***>,  Comment ***@***.***>
主题：Re: [mcordts/cityscapesScripts] How to test? (#143)
日期：2022年12月13日 00点27分







and how? Could you share the way to deal with it ？



give me ur email



Hi, Could you share the way to deal with it ？my email is ***@***.***,Thanks a lot.



&mdash;

**Comment by bohaoooooo** [comment]
Hello @Jess1989 , could you share the way how to translate to labelIDs ? My email is phchen.ai10@nycu.edu.tw,Thanks.

**Comment by Jess1989** [comment]
----- 原始邮件 -----
发件人：bohaoooooo ***@***.***>
收件人：mcordts/cityscapesScripts ***@***.***>
抄送人：Jess1989 ***@***.***>,  Mention ***@***.***>
主题：Re: [mcordts/cityscapesScripts] How to test? (#143)
日期：2022年12月25日 20点51分


Hello @Jess1989 , could you share the way how to translate to labelIDs ? My email is ***@***.***,Thanks.

**Comment by m828** [comment]
> > 又如何？你能分享一下处理它的方法吗？
> 
> 给我你的电子邮件

1040160736@qq,com，感谢🙇‍

**Comment by Jess1989** [comment]
----- 原始邮件 -----
发件人：m828 ***@***.***>
收件人：mcordts/cityscapesScripts ***@***.***>
抄送人：Jess1989 ***@***.***>,  Mention ***@***.***>
主题：Re: [mcordts/cityscapesScripts] How to test? (#143)
日期：2023年02月09日 17点17分







又如何？你能分享一下处理它的方法吗？



给我你的电子邮件



***@***.***,com，感谢🙇‍

reply：this is the file(trainIDS2LabelIDS)

**Comment by m828** [comment]
yes!




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "mcordts/cityscapesScripts"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年2月9日(星期四) 晚上6:36
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [mcordts/cityscapesScripts] How to test? (#143)





 
 ----- 原始邮件 -----
 发件人：m828 ***@***.***&gt;
 收件人：mcordts/cityscapesScripts ***@***.***&gt;
 抄送人：Jess1989 ***@***.***&gt;,  Mention ***@***.***&gt;
 主题：Re: [mcordts/cityscapesScripts] How to test? (#143)
 日期：2023年02月09日 17点17分
 
 
 
 
 
 
 
 又如何？你能分享一下处理它的方法吗？ 
 
 
 
 给我你的电子邮件 
 
 
 
 ***@***.***,com，感谢🙇‍
 
 reply：this is the file(trainIDS2LabelIDS)

**Comment by Jeremy-lf** [comment]
可以分享一下吗老哥？

**Comment by Jeremy-lf** [comment]
> yes!
> […](#)
> ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: "mcordts/cityscapesScripts" ***@***.***&gt;; 发送时间:&nbsp;2023年2月9日(星期四) 晚上6:36 ***@***.***&gt;; ***@***.******@***.***&gt;; 主题:&nbsp;Re: [mcordts/cityscapesScripts] How to test? (#143)
> ----- 原始邮件 ----- 发件人：m828 ***@***.***&gt; 收件人：mcordts/cityscapesScripts ***@***.***&gt; 抄送人：Jess1989 ***@***.***&gt;, Mention ***@***.***&gt; 主题：Re: [mcordts/cityscapesScripts] How to test? (#143) 日期：2023年02月09日 17点17分 又如何？你能分享一下处理它的方法吗？ 给我你的电子邮件 ***@***.***,com，感谢🙇‍ reply：this is the file(trainIDS2LabelIDS) — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you were mentioned.Message ID: ***@***.***&gt; — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you commented.Message ID: ***@***.***&gt;

可以分享一下吗老哥？

**Comment by finsary** [comment]
> > I have got 19 classes color png of testset,I want to know mIoU, how should I do ? Or how can I convert 19 cls into 34 cls to offical website?
> 
> have you solved this problem?

hi, can u help me how to test my cityscapes dataset locally, my email adress 379407282@qq.com, thx a lot!

**Comment by qqwewewejekl** [comment]
> > and how? Could you share the way to deal with it ？
> 
> give me ur email
442683978@qq.com thanks a lot!

**Comment by 156zg** [comment]
2334771821@qq.com thank a lot

**Comment by serdarch** [comment]
> I have got 19 classes color png of testset,I want to know mIoU, how should I do ? Or how can I convert 19 cls into 34 cls to offical website?

How can I have the colored .png files with fine annotations. Can you please help? I have to evaluate the network other than the format of the submission system.

Thanks in advance

**Comment by qqwewewejekl** [comment]
chatgpt教我正确使用cityscapes测试集 - 知乎 (zhihu.com)


吴马靖
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "mcordts/cityscapesScripts"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年3月13日(星期一) 下午3:44
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [mcordts/cityscapesScripts] How to test? (#143)





  
I have got 19 classes color png of testset,I want to know mIoU, how should I do ? Or how can I convert 19 cls into 34 cls to offical website?
  
How can I have the colored .png files with fine annotations. Can you please help? I have to evaluate the network other than the format of the submission system.
 
Thanks in advance

**Comment by Jess1989** [comment]
> 

give me ur email address

**Comment by qqwewewejekl** [comment]
442683978@qq.com

**Comment by serdarch** [comment]
> 又如何？你能分享一下处理它的方法吗？ 给我你的电子邮件



> How can I have the colored .png files with fine annotations. Can you please help? I have to evaluate the network other than the format of the submission system.
> 
> Thanks in advance

How can I have the colored .png files with fine annotations. Can you please help? I have to evaluate the network other than the format of the submission system.
my email serdarch@gmail.com
Thanks in advance

**Comment by Jess1989** [comment]
the way to translate the TrainIDS to LabelIDS is this file (https://github.com/Reagan1311/DABNet/blob/master/utils/trainID2labelID.py), then you need to compress the file to .zip.    now just upload the .zip file on the offical website(https://www.cityscapes-dataset.com/create-submission/) ,  remember to select the correct challenge category!

**Comment by Jess1989** [comment]
check the question 143, I have replied the question of test.
----- 原始邮件 -----
发件人：qqwewewejekl ***@***.***>
收件人：mcordts/cityscapesScripts ***@***.***>
抄送人：Jess1989 ***@***.***>,  Mention ***@***.***>
主题：Re: [mcordts/cityscapesScripts] How to test? (#143)
日期：2023年03月13日 16点14分


***@***.***

&mdash;

**Comment by Jess1989** [comment]
check the question
----- 原始邮件 -----
发件人：Serdar Erişen ***@***.***>
收件人：mcordts/cityscapesScripts ***@***.***>
抄送人：Jess1989 ***@***.***>,  Mention ***@***.***>
主题：Re: [mcordts/cityscapesScripts] How to test? (#143)
日期：2023年03月13日 16点14分



又如何？你能分享一下处理它的方法吗？ 给我你的电子邮件


How can I have the colored .png files with fine annotations. Can you please help? I have to evaluate the network other than the format of the submission system.
Thanks in advance

How can I have the colored .png files with fine annotations. Can you please help? I have to evaluate the network other than the format of the submission system.

my email ***@***.***

Thanks in advance

&mdash;

**Comment by serdarch** [comment]
Thanks for sharing the code, but I have already transformed labelIDs, I am
changing some labels (other than 19 categories) to see the improvement. So,
I need the annotated test images themselves in order to test the results
correctly. Submitting to the system does not help in my case.


13 Mar 2023 Pzt 11:21 tarihinde Jess1989 ***@***.***> şunu
yazdı:

**Comment by Jess1989** [comment]
the rgb image seems to be only useful for train set and val set, the test set cannot be convert to rgb form
----- 原始邮件 -----
发件人：Serdar Erişen ***@***.***>
收件人：mcordts/cityscapesScripts ***@***.***>
抄送人：Jess1989 ***@***.***>,  Mention ***@***.***>
主题：Re: [mcordts/cityscapesScripts] How to test? (#143)
日期：2023年03月13日 15点44分



I have got 19 classes color png of testset,I want to know mIoU, how should I do ? Or how can I convert 19 cls into 34 cls to offical website?

How can I have the colored .png files with fine annotations. Can you please help? I have to evaluate the network other than the format of the submission system.
Thanks in advance

&mdash;

**Comment by Jess1989** [comment]
> 

there is no annotated images of test set, only train set and val set can be used

**Comment by serdarch** [comment]
> I have got 19 classes color png of testset,I want to know mIoU, how should I do ? Or how can I convert 19 cls into 34 cls to offical website?

Can you please help how I can have color png of testset

**Comment by serdarch** [comment]
> > 
> 
> there is no annotated images of test set, only train set and val set can be used

Anyway, I thank you very much for your help!

**Comment by anizyu** [comment]
> > 
> 
> give me ur email address

1649006188@qq.com    thank you!

**Comment by finetune-master** [comment]
> > and how? Could you share the way to deal with it ？
> 
> give me ur email

2519676062@qq.com .Thanks a lot!
